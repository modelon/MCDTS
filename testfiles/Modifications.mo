package P

    package K
        constant Real k = 3;
    end K;
    
    function f
        input Real x;
        output Real y;
    algorithm
        y := x;
    end f;
    
    model A
        Real x;
    end A;
    
    model B
        Real x;
        Real y;
    end B;
    
    model C
        R r = R(0);
    end C;
    
    record R
        Real x;
    end R;
    
    record R2
        extends R(x = K.k);
    end R2;
    
    model D
        replaceable R r(x = f(0));
    end D;

    model M1 // Tests value modification
        A a(x = K.k);
    end M1;
    
    model M2 // Tests function call in value modification
        A a(x = f(0));
    end M2;
    
    model M3 // Tests multiple value modifications
        B b(x = K.k, y = f(0));
    end M3;
    
    model M4 // Tests nested value modification
        A a(x(min=K.k) = 0);
    end M4;
    
    model M5 // Tests value modification of type
        Real x(min = K.k);
    end M5;
    
    model M6 // Tests value modification of type
        Real x(min = f(0));
    end M6;
    
    model M7 // Tests function call in modification on extends
        extends A(x = f(0));
    end M7;
    
    model M8 // Tests function call in modification on component
        A a(x = f(0));
    end M8;
    
    model M9 // Tests record constructors in modifications
        C c(r = R2(0));
    end M9;
    
    model M10 // Tests redeclaring component with value modification
        extends D(redeclare R2 r);
    end M10;
    
    model M11 // Tests using a type that is modified in extends
        package A
            extends B(Temp(min = K.k));
        end A;
        
        package B
            type Temp = Real(min = 0);
            
            model M
                Temp T;
            end M;
        end B;
        
        A.M m;
    end M11;
    
    model M12 // Tests using a type that is modified in extends
        package A
            extends B(Temp(min = K.k));
            
            model C
                M m;
            end C;
        end A;
        
        package B
            type Temp = Real(min = 0);
            
            model M
                Temp T;
            end M;
        end B;
    end M12;
    
end P;
