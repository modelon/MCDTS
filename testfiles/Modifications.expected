P.K
P.f
P.B, P.K, P.f
P.A, P.K
P.K
P.f
P.f
P.f
P.R2, P.K
P.f
P.M11.A, P.K
P.M12.A, P.K